app/controllers/appointments_controller.rb

```diff
@@ -0,0 +1,75 @@
+class AppointmentsController < LoginController
+  before_action :confirm
+  before_action :set_appointment, only: %w[ show edit update destroy ]
+
+  # GET /appointments
+  # GET /appointments.json
+  def index
+    @appointments = Appointment.all
+  end
+
+  # GET /appointments/1
+  # GET /appointments/1.json
+  def show
+  end
+
+  # GET /appointments/new
+  def new
+    @appointment = Appointment.new
+  end
+
+  # GET /appointments/1/edit
+  def edit
+  end
+
+  # POST /appointments
+  # POST /appointments.json
+  def create
+    @appointment = Appointment.new(appointment_params)
+
+    respond_to do |format|
+      if @appointment.save
+        format.html { redirect_to @appointment, notice: "Appointment was successfully created." }
+        format.json { render :show, status: :created, location: @appointment }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @appointment.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /appointments/1
+  # PATCH/PUT /appointments/1.json
+  def update
+    respond_to do |format|
+      if @appointment.update(appointment_params)
+        format.html { redirect_to @appointment, notice: "Appointment was successfully updated." }
+        format.json { render :show, status: :ok, location: @appointment }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @appointment.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /appointments/1
+  # DELETE /appointments/1.json
+  def destroy
+    @appointment.destroy
+    respond_to do |format|
+      format.html { redirect_to appointments_url, notice: "Appointment was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_appointment
+      @appointment = Appointment.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def appointment_params
+      params.fetch(:appointment, {})
```

Comment:
from the comment it seems like you want to only allow certain parameters through, but fetch alone doesn't achieve this. You will need something like `params.fetch(:appointment, {}).permit(:attributes, :you, :want, :to, :allow)`

Reply:
I don't know what parameters are needed yet. Maybe we can leave this function for now and make modifications to it in the future?